### PR TITLE
[CI] Bump CUDA in Docker images to 11.6.1

### LINKED ISSUE #14347: Failing (flaky?) test in GPU CI
## 🐛 Bug

```python
>           work = group.broadcast([tensor], opts)
E           RuntimeError: Input tensor data type is not supported for NCCL process group: BFloat16

/home/AzDevOps_azpcontainer/.local/lib/python3.9/site-packages/torch/distributed/distributed_c10d.py:1197: RuntimeError
----------------------------- Captured stderr call -----------------------------
initializing deepspeed distributed: GLOBAL_RANK: 1, MEMBER: 2/2
LOCAL_RANK: 1 - CUDA_VISIBLE_DEVICES: [0,1]
=========================== short test summary info ============================
FAILED strategies/test_deepspeed_strategy.py::test_deepspeed_with_bfloat16_precision
```

### To Reproduce

https://dev.azure.com/Lightning-AI/lightning/_build/results?buildId=90723&view=logs&j=3f274fac-2e11-54ca-487e-194c91f3ae9f&t=7ba6ccc2-87f1-5d8a-85e1-a52b03435dd6&l=1003

### Expected behavior

Green CI

### Environment

Current master

cc @tchaton @rohitgr7 @carmocca @akihironitta @borda

### PR DESCRIPTION
**Pull request by otaj**
### Bump CUDA in Docker images to 11.6.1 in order to have more recent NCCL version

Fixes #14347 

## Before submitting

- [x] Was this **discussed/approved** via a GitHub issue? (not for typos and docs)
- [x] Did you read the [contributor guideline](https://github.com/Lightning-AI/lightning/blob/master/.github/CONTRIBUTING.md), **Pull Request** section?
- [x] Did you make sure your **PR does only one thing**, instead of bundling different changes together?
- [x] Did you make sure to **update the documentation** with your changes? (if necessary)
- [x] Did you write any **new necessary tests**? (not for typos and docs)
- [x] Did you verify new and **existing tests pass** locally with your changes?
- [x] Did you list all the **breaking changes** introduced by this pull request?
- [n/a] Did you **update the CHANGELOG**? (not for typos, docs, test updates, or minor internal changes/refactors)

## PR review

Anyone in the community is welcome to review the PR.
Before you start reviewing, make sure you have read the [review guidelines](https://github.com/Lightning-AI/lightning/wiki/Review-guidelines). In short, see the following bullet-list:

- [x] Is this pull request ready for review? (if not, please submit in draft mode)
- [x] Check that all items from **Before submitting** are resolved
- [x] Make sure the title is self-explanatory and the description concisely explains the PR
- [x] Add labels and milestones (and optionally projects) to the PR so it can be classified

## Did you have fun?

Make sure you had fun coding 🙃

cc @tchaton @rohitgr7 @carmocca @akihironitta @borda

### CODE REVIEW

File: .github/workflows/ci-pytorch-dockers.yml [hunk 1]
```diff
@@ -1,28 +1,14 @@
 name: Docker
 
 on:
-  push:
```
Comment (Borda) [inline .github/workflows/ci-pytorch-dockers.yml:4]: Is this intended?

Comment (otaj) [reply .github/workflows/ci-pytorch-dockers.yml:4]: It's taken from here (https://github.com/Lightning-AI/lightning/pull/13716/commits/11c58e01d4813e740bae04ca2f4f3cee47c3cc8a), I fully intend to revert that commit

File: dockers/base-cuda/Dockerfile [hunk 2]
```diff
@@ -13,7 +13,7 @@
 # limitations under the License.
 
 ARG UBUNTU_VERSION=20.04
-ARG CUDA_VERSION=11.3.1
+ARG CUDA_VERSION=11.6.1
```
Comment (carmocca) [inline dockers/base-cuda/Dockerfile:16]: The cuda image build is blocked by the lack of a bagua wheel for cuda 11.6.1

`#13 0.911 ERROR: No matching distribution found for bagua-cuda116==0.9.0`

Maybe bagua should still use the old cuda version? Or is there a plan to release an updated one? cc @wangraying

File: dockers/base-conda/Dockerfile [hunk 3]
```diff
@@ -73,9 +74,10 @@ COPY environment.yml environment.yml
 # conda init
 RUN \
     conda update -n base -c defaults conda && \
+    CUDA_VERSION_MM=$(python -c "print('.'.join('$CUDA_VERSION'.split('.')[:2]))") && \
     conda create -y --name $CONDA_ENV \
-      python=${PYTHON_VERSION} pytorch=${PYTORCH_VERSION} torchvision torchtext cudatoolkit=${CUDA_VERSION} \
-      -c nvidia -c pytorch -c pytorch-test -c pytorch-nightly && \
```
Comment (Borda) [inline dockers/base-conda/Dockerfile:78]: Why is this removed? it serves for sourcing future test or nightly versions...

Comment (otaj) [reply dockers/base-conda/Dockerfile:78]: We're pinning the version of pytorch anyway, so `pytorch-nightly` is definitely never going to be used (it has versions like `1.13.0.dev`. `pytorch-test` seems like a duplicate of `pytorch` channel and they looked obsolete when pinning versions anyway.

Comment (otaj) [reply dockers/base-conda/Dockerfile:78]: If you want, I can add them back, but then the building of the images will take another hour

**Review by awaelchli — APPROVED** [review]

File: .github/workflows/ci-pytorch-dockers.yml (hunk 1 above)
Comment (otaj) [reply .github/workflows/ci-pytorch-dockers.yml]: Reverted now

File: dockers/base-cuda/Dockerfile (hunk 2 above)
Comment (otaj) [reply dockers/base-cuda/Dockerfile]: According to their docs, `bagua-cuda115` should work with all `CUDA_VERSION >= 11.5`. I tried it and it seems to build

**Review by rohitgr7 — APPROVED** [review]
thank you Ota!

**Review by kaushikb11 — APPROVED** [review]

File: dockers/base-conda/Dockerfile [hunk 4]
```diff
@@ -73,9 +74,10 @@ COPY environment.yml environment.yml
...
+      python=${PYTHON_VERSION} pytorch=${PYTORCH_VERSION} torchvision torchtext cudatoolkit=${CUDA_VERSION_MM} \
+      -c nvidia -c pytorch  && \
```
Comment (Borda) [inline dockers/base-conda/Dockerfile:80]:
```suggestion
      -c nvidia -c pytorch -c pytorch-test && \
```

**Review by Borda — APPROVED** [review]

**Review by carmocca — APPROVED** [review]

**Pull request merged**